loom/graphics/gfxVectorRenderer.cpp

```diff
@@ -1,544 +1,546 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ===========================================================================
+ */
+
+#include <string.h>
+#include "stdio.h"
+
+#include "loom/common/core/log.h"
+#include "loom/common/core/allocator.h"
+#include "loom/common/core/assert.h"
+#include "loom/common/assets/assets.h"
+
+#include "loom/graphics/gfxMath.h"
+#include "loom/graphics/gfxGraphics.h"
+#include "loom/graphics/gfxVectorRenderer.h"
+
+#include "nanovg.h"
+
+#ifdef LOOM_RENDERER_OPENGLES2
+#define NANOVG_GLES2_IMPLEMENTATION
+#else
+#define NANOVG_GL2_IMPLEMENTATION
+#endif
+
+#include "nanovg_lm_gl.h"
+#include "nanovg_lm_gl_utils.h"
+
+#define NANOSVG_IMPLEMENTATION
+#include "nanosvg.h"
+
+/*
+#include <windows.h>
+#include <wchar.h>
+
+#if defined(_MSC_VER) && _MSC_VER > 1310
+// Visual C++ 2005 and later require the source files in UTF-8, and all strings 
+// to be encoded as wchar_t otherwise the strings will be converted into the 
+// local multibyte encoding and cause errors. To use a wchar_t as UTF-8, these 
+// strings then need to be convert back to UTF-8. This function is just a rough 
+// example of how to do this.
+# define utf8(str)  ConvertToUTF8(L##str)
+const char * ConvertToUTF8(const wchar_t * pStr) {
+	static char szBuf[1024];
+	WideCharToMultiByte(CP_UTF8, 0, pStr, -1, szBuf, sizeof(szBuf), NULL, NULL);
+	return szBuf;
+}
+#else
+// Visual C++ 2003 and gcc will use the string literals as is, so the files 
+// should be saved as UTF-8. gcc requires the files to not have a UTF-8 BOM.
+# define utf8(str)  str
+#endif
+*/
+
+namespace GFX
+{
+lmDefineLogGroup(gGFXVectorRendererLogGroup, "GFXVectorRenderer", 1, LoomLogInfo);
+
+NVGcontext *nvg = NULL;
+static int font;
+
+utHashTable<utHashedString, utString> VectorTextFormat::loadedFonts;
+int VectorRenderer::frameWidth = 0;
+int VectorRenderer::frameHeight = 0;
+uint8_t VectorRenderer::quality = VectorRenderer::QUALITY_ANTIALIAS | VectorRenderer::QUALITY_STENCIL_STROKES;
+
+//*
+void drawLabel(struct NVGcontext* vg, const char* text, float x, float y, float w, float h)
+{
+	NVG_NOTUSED(w);
+
+	nvgFontSize(vg, 30.0f);
+	nvgFontFace(vg, "sans");
+	nvgFillColor(vg, nvgRGBA(0, 25, 25, 128));
+
+	nvgTextAlign(vg, NVG_ALIGN_LEFT | NVG_ALIGN_MIDDLE);
+	nvgText(vg, x, y + h*0.5f, text, NULL);
+}
+//*/
+
+void VectorRenderer::setSize(int width, int height) {
+	frameWidth = width;
+	frameHeight = height;
+}
+
+void VectorRenderer::beginFrame()
+{
+    nvgBeginFrame(nvg, frameWidth, frameHeight, 1);
+
+
+
+    /*
+    nvgBeginPath(nvg);
```

Comment:
Oops, looks like I committed weird whitespace at some point. That file is messy, but for another PR I think.